lib/snapshot-manager.js

```diff
@@ -388,23 +373,20 @@ function determineSnapshotDir(options) {
 
 function determineSourceMappedDir(options) {
 	const sourceMapFilePath = path.join(options.testDir, `${options.name}.map`);
-	const sourceMapContent = tryReadSourceMap(sourceMapFilePath);
-
-	if (sourceMapContent) {
-		const targetFilename = path.basename(sourceMapContent.file, 'js');
-		const testFileSource = sourceMapContent.sources.find(file => {
-			const fileName = file.replace(path.extname(targetFilename), '');
-			return fileName.indexOf(targetFilename) > -1;
-		});
-		if (testFileSource) {
-			if (options.location) {
-				const basePath = path.join(options.projectDir, options.location);
-				const sourceTestFilePath = path.dirname(path.resolve(options.testDir, testFileSource));
-				const relativePath = sourceTestFilePath.replace(options.projectDir, '');
-				return path.join(basePath, relativePath);
-			}
-			return path.dirname(path.resolve(options.testDir, testFileSource));
-		}
+	const sourceMapContent = tryRead(options.relFile).toString();
```

Comment:
`tryRead()` returns `null` if the file doesn't exist. You'll have to read it into a variable and bail out of if it's `null`.

Perhaps this function should return `null` if no source-mapped directory could be determined. Then `determineSnapshotDir()` can do `testDir = determineSourceMappedDir(options) || options.testDir`.